### PR TITLE
[CHAT-1469] Add support for pinned messages

### PR DESCRIPTION
**Pull request by AnatolyRugalev**
This PR adds support for message pinning.

The supposed way to query pinned messages is: 

```js
channel.search({pinned: true});
```

The Message object in API gets new fields:

* pinned: boolean
* pinned_at: string
* pin_expires: string
* pinned_by: UserResponse

To pin or unpin message client should set `pinned` and `pin_expires` to desired values.

```js
// create pinned message
const {message} = await channel.sendMessage({
  text: "my message",
  pinned: true,
  pin_expires: "2077-01-01T00:00:00Z",
});

// unpin message
await channel.unpinMessage(message);

// pin message for 120 seconds
client.pinMessage(message, 120);

// change message expiration to 2077
await client.pinMessage(message, "2077-01-01T00:00:00Z");

// remove expiration date from pinned message
client.pinMessage(message);
```

### CODE REVIEW

**Review by ferhatelmas — COMMENTED** [review]
LGTM so far

how about testing permissions ?

**Comment by AnatolyRugalev** [comment]
@ferhatelmas done

File: test/integration/messages.js [hunk 1]
```diff
@@ -0,0 +1,281 @@
...
+			const { message } = await chat.owner.channel.sendMessage({
+				text: 'Pinned message 2',
+				pinned: true,
+			});
+			expect(message.pinned).to.be.equal(true);
+			const { message: updatedMessage } = await chat.owner.client.updateMessage({
+				id: message.id,
+				pinned: false,
+			});
+			expect(updatedMessage.pinned).to.be.equal(false);
```
Comment (thesyncim) [inline test/integration/messages.js:160]: can you check if the message text persists after unpin?

Comment (thesyncim) [reply test/integration/messages.js:160]: time to add partial message update :)

Comment (ferhatelmas) [reply test/integration/messages.js:160]: makes sense actually :sweat_smile:

Comment (AnatolyRugalev) [reply test/integration/messages.js:161]: Yup, missed that. I've fixed the tests, but the partial update would be a topic for discussion in a different PR I think

File: README.md [hunk 2]
```diff
@@ -138,13 +138,16 @@ Custom types are carried into all creation functions as well.
...
+  'TestToken',
+);
 
 // Invalid
-client.setUser({ id: 'testId' }, 'TestToken'); // Type ChatUser1 | ChatUser2 requires nickname for both types
-client.setUser({ id: 'testId', nickname: true }, 'TestToken'); // nickname must be a string
-client.setUser({ id: 'testId', nickname: 'testUser', country: 'NL' }, 'TestToken'); // country does not exist on type ChatUser1 | ChatUser2
+client.connectUser({ id: 'testId' }, 'TestToken'); // Type ChatUser1 | ChatUser2 requires nickname for both types
+client.connectUser({ id: 'testId', nickname: true }, 'TestToken'); // nickname must be a string
+client.connectUser({ id: 'testId', nickname: 'testUser', country: 'NL' }, 'TestToken'); // country does not exist on type ChatUser1 | ChatUser2
```
Comment (ferhatelmas) [inline README.md:150]: this is nice but might want to split it and update all occurrences in the repo

File: test/integration/utils.js [hunk 3]
```diff
@@ -9,7 +11,8 @@ const apiSecret = process.env.STREAM_API_SECRET;
 export function getTestClient(serverSide) {
 	return new StreamChat(apiKey, serverSide ? apiSecret : null, {
 		timeout: 15000,
-		allowServerSideConnect: serverSide,
+		// mute warnings because tests are being run on server-side
```
Comment (ferhatelmas) [inline test/integration/utils.js:14]: :+1:

File: test/integration/messages.js [hunk 4]
```diff
@@ -0,0 +1,284 @@
...
+Promise.config({
+	longStackTraces: true,
+	warnings: {
+		wForgottenReturn: false,
+	},
+});
+
+// setupTestChannel sets up all possible client and channel objects for given channel type
+// set createOnServerSide=true to execute channel creation from the server side
+async function setupTestChannel(type, createOnServerSide, custom = {}) {
```
Comment (ferhatelmas) [inline test/integration/messages.js:30]: This is very nice and makes testing declarative and simple.

**Review by ferhatelmas — APPROVED** [review]
LGTM

**Comment by AnatolyRugalev** [comment]
I'm going to extract `connectUser` stuff into separate PR

**Comment by ferhatelmas** [comment]
> I'm going to extract connectUser stuff into separate PR

sounds better, thanks

File: src/types.ts [hunk 5]
```diff
@@ -1467,10 +1467,14 @@ export type MessageBase<
   UserType = UnknownType
 > = MessageType & {
   id: string;
+  pinned: boolean;
```
Comment (DanC5) [inline src/types.ts:1470]: @AnatolyRugalev will pinned always exist on a message object? So if a message is sent without specifying pinned: true will it default to false?

Comment (AnatolyRugalev) [reply src/types.ts:1470]: Yes

**Review by ferhatelmas — APPROVED** [review]
LGTM 

nice convenience helpers :+1:

**Pull request merged**